thingsboard_gateway/connectors/modbus_async/modbus_connector.py

```diff
@@ -0,0 +1,697 @@
+                response = await slave.read(config['functionCode'], config['address'], config['objectsCount'])
+
+                if 'Exception' in str(response) or 'Error' in str(response):
+                    self.__log.error("Reading failed for device %s function code %s address %s unit id %s",
+                                     slave.device_name, config['functionCode'], config[ADDRESS_PARAMETER],
+                                     slave.unit_id)
+                    self.__log.error("Reading failed with exception:", exc_info=result)
+                    self.__log.info("Trying to reconnect to device %s", slave.device_name)
+                    if slave.master.connected:
+                        await slave.master.close()
+                    await slave.connect()
+                    if slave.master.connected:
+                        self.__log.info("Reconnected to device %s", slave.device_name)
+                        response = await slave.read(config['functionCode'], config['address'], config['objectsCount'])
+                        if "Exception" in str(result) or "Error" in str(result):
+                            self.__log.error("Reading failed for device %s function code %s address %s unit id %s",
+                                             slave.device_name, config['functionCode'], config[ADDRESS_PARAMETER],
+                                             slave.unit_id)
+                            self.__log.error("Reading failed with exception:", exc_info=result)
+
+                            if slave.master.connected:
+                                await slave.master.close()
+
+                            self.__log.info("Will try to connect to device %s later", slave.device_name)
+
+                result[config_section][config['tag']] = response
+
+        return result
+
+    def __manage_device_connectivity_to_platform(self, slave: Slave):
+        if slave.master.connected and not slave.is_connected_to_platform():
+            self.__add_device_to_platform(slave)
+        else:
+            self.__delete_device_from_platform(slave)
+
+    def __delete_device_from_platform(self, slave: Slave):
+        self.__gateway.del_device(slave.device_name)
+        slave.last_connect_time = 0
+
+    def __add_device_to_platform(self, slave: Slave):
+        """
+        Add device to platform
+        """
+
+        device_connected = slave.is_connected_to_platform()
+        if not device_connected and slave.master.connected:
+            device_connected = self.__gateway.add_device(slave.device_name,
+                                                         {CONNECTOR_PARAMETER: self},
+                                                         device_type=slave.device_type)
+
+            slave.last_connect_time = monotonic() if device_connected else 0
+
+    def __convert_data(self):
+        while not self.__stopped:
+            batch_to_convert = {}
```

Comment:
Let’s add try…except to `while not self.stopped` loop and add exceptions handling.